src/docs.js

```diff
@@ -6,38 +6,48 @@ let path = require('path');
 let recursiveReadSync = require('recursive-readdir-sync');
 
 async function documenter(options) {
-
   options = _.defaults({}, options, {
+    metadata: null,
     schemas: [],
     docsFolder: null,
+    references: [],
   });
   assert(options.schemas, 'options.schemas must be given');
   assert(options.schemas instanceof Array, 'options.schemas must be an array');
 
-  let schemas = options.schemas;
   let tarball = tar.pack();
+  let metadata = options.metadata;
+
+  if (metadata) {
```

Comment:
if each tarball is a documentation section (with possible subsections)...

How do you insert them all in a menu?

You can order them alphabetically... or we could have each include a menu
index... if we start using integers like 1000, 2000, 3000, then we can
always create a section like 1500 that goes in between the two...

I assume the sitemap will look somewhat like this:

queue
   bla-bla-link   (some markdown file in the tarball from queue)
   ...
   APIs             (api references in the tarball from queue)
   Exchanges   (exchange references in the tarball from queue)
auth
   ...
index
   ...

Docs from queue, auth, index arrives in different tarballs. So how do you
order them into a menu... Using menuIndex.
It's not perfect, but at-least it gives us some control... All though any
project could decide that it wants menuIndex:0 and hence, end up at the
top...
But IMO that not a huge security risk, as we sort of somewhat trust most of
the projects that submit documentation :)



--
Regards Jonas Finnemann Jensen.

2016-07-20 11:29 GMT-07:00 kristelteng <notifications@github.com>:

> In src/docs.js
> <https://github.com/taskcluster/taskcluster-lib-docs/pull/7#discussion_r71580576>
> :
>
> >    let tarball = tar.pack();
> > +  let metadata = options.metadata;
> > +
> > +  if (metadata) {
>
> @jonasfj <https://github.com/jonasfj> : Can you please explain what is
> menuIndex again?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/taskcluster/taskcluster-lib-docs/pull/7/files/711349dee3efe9d99d0868d117b38ead2853f2e4#r71580576>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAJI5JKToVwHekl_bcCiRBbNu_wKmUN8ks5qXmkPgaJpZM4JOV19>
> .
>